src/debugpy/adapter/clients.py

```diff
@@ -701,6 +702,24 @@ def disconnect_request(self, request):
     def disconnect(self):
         super().disconnect()
 
+    def report_sockets(self):
+        sockets = [
+            {
+                "host": host,
+                "port": port,
+                "internal": listener is not clients.listener,
+            }
+            for listener in [clients.listener, launchers.listener, servers.listener]
```

Comment:
What's going on with the lack of indenting in this for loop?

Reply:
This is a Python list comprehension. C# equivalent would be something like:
```cs
from listener in ...
where listener is ...
let (host, port) = listener.getsockname()
select {...}
.ToList()
```